recipes-bsp/firmware-imx/firmware-imx_8.6.bb

```diff
@@ -0,0 +1,118 @@
+    file://sdma \
+    file://epdc \
+    file://regulatory \
+    file://hdmi \
+    file://sdma-firmware.service \
+    file://epdc-firmware.service \
+    file://regulatory-firmware.service \
+    file://hdmi-firmware.service \
+"
+
+PE = "1"
+
+inherit allarch systemd
+
+SYSTEMD_PACKAGES = "${PN}-sdma ${PN}-epdc ${PN}-regulatory ${PN}-hdmi"
+SYSTEMD_SERVICE_${PN}-sdma = "sdma-firmware.service"
+SYSTEMD_SERVICE_${PN}-epdc = "epdc-firmware.service"
+SYSTEMD_SERVICE_${PN}-regulatory = "regulatory-firmware.service"
+SYSTEMD_SERVICE_${PN}-hdmi = "hdmi-firmware.service"
+
+do_install() {
+    install -d ${D}${base_libdir}/firmware/imx
+
+    if ${@bb.utils.contains('DISTRO_FEATURES','systemd','true','false',d)}; then
+        install -d ${D}${sysconfdir}
+        install -d ${D}${systemd_system_unitdir}
+        install -m 0755 ${WORKDIR}/sdma ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/epdc ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/regulatory ${D}${sysconfdir}
+        install -m 0755 ${WORKDIR}/hdmi ${D}${sysconfdir}
+        install -m 0644 ${WORKDIR}/sdma-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/epdc-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/regulatory-firmware.service ${D}${systemd_system_unitdir}
+        install -m 0644 ${WORKDIR}/hdmi-firmware.service ${D}${systemd_system_unitdir}
+    fi
+
+    cd firmware
+    for d in *; do
+        case $d in
+        ddr|seco)
+            # These folders are for i.MX 8 and are included in the boot image via imx-boot
+            bbnote Excluding folder $d
+            ;;
+        *)
+            cp -rfv $d ${D}${base_libdir}/firmware
+            ;;
+        esac
+    done
+    cd -
+
+    # Install SDMA Firmware: sdma-imx6q.bin & sdma-imx7d.bin into lib/firmware/imx/sdma
+    install -d ${D}${base_libdir}/firmware/imx/sdma
+    mv ${D}${base_libdir}/firmware/sdma/sdma-imx6q.bin ${D}${base_libdir}/firmware/imx/sdma
+    mv ${D}${base_libdir}/firmware/sdma/sdma-imx7d.bin ${D}${base_libdir}/firmware/imx/sdma
+
+    mv ${D}${base_libdir}/firmware/epdc/ ${D}${base_libdir}/firmware/imx/epdc/
+    mv ${D}${base_libdir}/firmware/imx/epdc/epdc_ED060XH2C1.fw.nonrestricted ${D}${base_libdir}/firmware/imx/epdc/epdc_ED060XH2C1.fw
+
+    mv ${D}${base_libdir}/firmware/easrc/ ${D}${base_libdir}/firmware/imx/easrc/
+
+    # Install HDMI Firmware: hdmitxfw.bin, hdmirxfw.bin & dpfw.bin into lib/firmware/imx/hdmi
+    install -d ${D}${base_libdir}/firmware/imx/hdmi
+    mv ${D}${base_libdir}/firmware/hdmi/cadence/hdmitxfw.bin ${D}${base_libdir}/firmware/imx/hdmi
+    mv ${D}${base_libdir}/firmware/hdmi/cadence/hdmirxfw.bin ${D}${base_libdir}/firmware/imx/hdmi
+    mv ${D}${base_libdir}/firmware/hdmi/cadence/dpfw.bin ${D}${base_libdir}/firmware/imx/hdmi
+
+    find ${D}${base_libdir}/firmware -type f -exec chmod 644 '{}' ';'
+    find ${D}${base_libdir}/firmware -type f -exec chown root:root '{}' ';'
```

Comment:
Why is this required?

And if you use `find` - please use `+` instead of `;` for performance reasons... 😉

Reply:
Not sure, it's inherited. I wasn't focus on this